internal/daemon/cluster/handlers/worker_service.go

```diff
@@ -48,14 +50,16 @@ func NewWorkerServiceServer(
 	connectionRepoFn common.ConnectionRepoFactory,
 	updateTimes *sync.Map,
 	kms *kms.Kms,
+	livenessTimeToStale *atomic.Int64,
 ) *workerServiceServer {
 	return &workerServiceServer{
-		serversRepoFn:    serversRepoFn,
-		workerAuthRepoFn: workerAuthRepoFn,
-		sessionRepoFn:    sessionRepoFn,
-		connectionRepoFn: connectionRepoFn,
-		updateTimes:      updateTimes,
-		kms:              kms,
+		serversRepoFn:       serversRepoFn,
+		workerAuthRepoFn:    workerAuthRepoFn,
+		sessionRepoFn:       sessionRepoFn,
+		connectionRepoFn:    connectionRepoFn,
+		updateTimes:         updateTimes,
+		kms:                 kms,
+		livenessTimeToStale: livenessTimeToStale,
```

Comment:
if `livenessTimeToStale` is nil, there's a possible panic when this is accessed.

Reply:
It's true. I didn't want to change this to return an error right now; I'm relying on the fact that we know it will not be nil for any real controller/worker (since it's explicitly set in the New functions) so only tests might panic and then we can and should fix the test.